custom-metadata-provider-specification.yaml

```diff
@@ -0,0 +1,124 @@
+openapi: 3.0.0
+servers: 
+  - url: https://example.com
+    description: Local server 
+info:
+  license: 
+    name: MIT
+    url: https://opensource.org/licenses/MIT
+
+    
+  title: Custom Metadata Provider
+  version: 0.1.0
+security:
+  - api_key: []
+
+paths:
+  /search:
+    get:
+      description: Search for books
+      operationId: search
+      summary: Search for books
+      security: 
+        -  api_key: []
+      parameters:
+        - name: query
+          in: query
+          required: true
+          schema:
+            type: string
+        - name: author
+          in: query
+          required: false
+          schema:
+            type: string
+      responses:
+        "200":
+          description: OK
+          content:
+            application/json:
+              schema:
+                type: object
+                properties:
+                  matches:
+                    type: array
+                    items:
+                      $ref: "#/components/schemas/BookMetadata"
+        "400":
+          description: Bad Request
+          content:
+            application/json:
+              schema:
+                type: object
+                properties:
+                  error:
+                    type: string
+        "401":
+          description: Unauthorized
+          content:
+            application/json:
+              schema:
+                type: object
+                properties:
+                  error:
+                    type: string
+        "500":
+          description: Internal Server Error
+          content:
+            application/json:
+              schema:
+                type: object
+                properties:
+                  error:
+                    type: string
+components:
+  schemas:
+    BookMetadata:
+      type: object
+      properties:
+        title:
+          type: string
+        subtitle:
+          type: string
+        author:
+          type: string
+        narrator:
+          type: string
+        publisher:
+          type: string
+        published_year:
```

Comment:
Should we use camelCase here to stay consistent with our data models? publishedYear